lib/streamlit/elements/heading.py

```diff
@@ -161,14 +160,32 @@ def title(
 
         """
         title_proto = HeadingProto()
-        if anchor is not None:
-            title_proto.anchor = anchor
-        title_proto.body = clean_text(body)
-        title_proto.tag = TITLE_TAG
-
+        marshall(title_proto, body=body, anchor=anchor, tag=TITLE_TAG)
         return self.dg._enqueue("heading", title_proto)
 
     @property
     def dg(self) -> "DeltaGenerator":
         """Get our DeltaGenerator."""
         return cast("DeltaGenerator", self)
+
+
+def marshall(
+    heading_proto: HeadingProto,
+    *,
+    body,
+    anchor: Union[None, str, Literal[False]],
+    tag: str,
+):
+    """Marshall heading's data into proto."""
+    if anchor is not None:
+        if anchor is False:
+            heading_proto.hide_anchor = True
+        elif isinstance(anchor, str):
+            heading_proto.anchor = anchor
+        else:
+            raise StreamlitAPIException(
+                "Anchor parameter has invalid value: %s" % anchor
```

Comment:
Nit: this may be misleading. What if I want my anchor text to be `"6"` but I provide it as an integer? I'll get information about incorrect value where in fact an incorrect type is a  real source of an error.

Reply:
I updated the value validation. Now we have 2 separate error messages.